NOTES/HowStuffWorks.md

```diff
@@ -0,0 +1,84 @@
+# How Stuff Works
+This documentation is for grasping the overall design. No details by intention. To get up and running quickly see `/README.md`.
+
+At first sight, it may not be immediately clear why/how certain things are called/triggered.
+The below explanation points you in the right direction, so you can research further.
+Some (inconspicuously) used/relevant node modules are explicitly mentioned with package name.
+
+*****
+
+## How the app is served.
+A JS webserver is included in the form of `/lib/server.js`. It can be run in two modes:
+
+### development mode
+Run the server using `npm run start-dev`.
+It wil use `/lib/server-development.js` which wil use `/lib/server.js` which wil use `/config/simple.js` which wil use `/app/simple.html`. A tiny HTML file is loaded, the JS is downloaded + executed + rendered, and ultimately output is shown.
+
+### production mode
+In this mode, the React HTML output is (pre)rendered (and populated) on the server (a.k.a. isomorphic).
+
+Run the server using `npm run start`.
+It wil use `/lib/server-production.js` which wil use `/lib/server.js` which wil use `/config/prerender.jsx` which wil use `/app/prerender.html`. A big HTML file is loaded (and output is shown immediately), the JS is downloaded + executed + rendered, and output is updated.
+
+This server side render (prerender) is possible because the React JS can be executed on the server.
+In your browser the main `React.render(<Application />, document.getElementById("content"))` call (in `/config/app.jsx`) outputs to the browser DOM.
+But when prerendering, the main `React.renderToString(<Application />)` call (in `/app/prerender.jsx`) outputs to a string, which is inserted into the HTML (including React component states) as content.
```

Comment:
Good one thanks. I'll make a "How the build works" with webpack train-of-thought. I'm going to include some questions to you, parts are not entirely clear to me yet.